src/routes/quiz/[collection]/[id]/+page.js

```diff
@@ -1,21 +1,58 @@
 import { base } from '$app/paths';
+import config from '$lib/data/config';
+import { checkQuizAccess } from '$lib/data/quiz';
 
 /** @type {import('./$types').PageLoad} */
 export async function load({ params, fetch }) {
     const id = params.id;
     const collection = params.collection;
 
-    try {
-        const response = await fetch(`${base}/collections/${collection}/quizzes/${id}.json`);
+    const bookCollection = config.bookCollections.find((x) => x.id === collection);
+    const book = bookCollection.books.find((x) => x.id === id);
 
-        if (!response.ok) {
-            throw new Error('Failed to fetch quiz JSON file');
-        }
+    let locked = false;
+    let dependentQuizId = null;
+    let dependentQuizName = null;
+
+    if (book.quizFeatures['access-type'] === 'after') {
+        dependentQuizId = book.quizFeatures['access-after'];
+        const accessGranted = await checkQuizAccess(dependentQuizId);
+        locked = !accessGranted;
+    }
+
+    if (locked && dependentQuizId) {
+        const dependentBook = bookCollection.books.find((x) => x.id === dependentQuizId);
+        dependentQuizName = dependentBook ? dependentBook.name : dependentQuizId;
+    }
+
+    if (!locked) {
+        try {
+            const response = await fetch(`${base}/collections/${collection}/quizzes/${id}.json`);
+            if (!response.ok) {
+                throw new Error('Failed to fetch quiz JSON file');
+            }
 
-        const quizData = await response.json();
-        return { quiz: quizData };
-    } catch (error) {
-        console.error('Error fetching quiz JSON file:', error);
-        return {};
+            const quizData = await response.json();
+            return {
+                quiz: quizData,
+                locked,
+                quizId: id,
+                quizName: book.name,
+                dependentQuizId,
+                dependentQuizName,
+                passScore: quizData.passScore
+            };
+        } catch (error) {
+            console.error('Error fetching quiz JSON file:', error);
+        }
     }
+
+    return {
+        locked,
+        quizId: id,
+        quizName: book.name,
+        dependentQuizId,
+        dependentQuizName,
+        passScore: 0
```

Comment:
Why returning "passScore: 0" ?

Reply:
If it is locked then the passScore doesn't matter?  Why is there two separate returns?

Reply:
I have fixed to have a single return statement